index.js

```diff
@@ -390,21 +390,33 @@ Client.prototype._encode = function (obj, enc) {
 }
 
 Client.prototype.sendSpan = function (span, cb) {
+  if (this.destroyed) {
+    return
+  }
   this._maybeCork()
   return this.write({ span }, Client.encoding.SPAN, cb)
 }
 
 Client.prototype.sendTransaction = function (transaction, cb) {
+  if (this.destroyed) {
+    return
+  }
   this._maybeCork()
   return this.write({ transaction }, Client.encoding.TRANSACTION, cb)
 }
 
 Client.prototype.sendError = function (error, cb) {
+  if (this.destroyed) {
+    return
+  }
   this._maybeCork()
   return this.write({ error }, Client.encoding.ERROR, cb)
 }
 
 Client.prototype.sendMetricSet = function (metricset, cb) {
+  if (this.destroyed) {
```

Comment:
nit: Would we be better off with a helper method for these checks since they're the same check for the same reason? -- Something like `Client.prototype._isSafeToWrite`?

Reply:
Does https://github.com/elastic/apm-nodejs-http-client/pull/137/commits/1ec1efa2cf12be6750fdba709af37d96ffa01064 look reasonable?